testing/matrix-sdk-test/src/lib.rs

```diff
@@ -1,18 +1,54 @@
 use http::Response;
 pub use matrix_sdk_test_macros::async_test;
-use ruma::api::{client::sync::sync_events::v3::Response as SyncResponse, IncomingResponse};
+use once_cell::sync::Lazy;
+use ruma::{
+    api::{client::sync::sync_events::v3::Response as SyncResponse, IncomingResponse},
+    user_id, UserId,
+};
 use serde_json::Value as JsonValue;
 
+/// Create a `Raw<AnyTimelineEvent>` from arbitrary JSON.
+///
+/// Forwards all arguments to [`serde_json::json`].
+#[macro_export]
+macro_rules! timeline_event {
```

Comment:
Small suggestion: could we name this `any_timeline_event`? I'm always confused about all the different timeline event types, so making it very explicit which it is would be helpful to me.

Reply:
Ah but it returns a raw event, which can truly contain arbitrary events, not just the ones Ruma knows 😅